Model/Product/Builder.php

```diff
@@ -277,6 +277,9 @@ public function build(
                     $this->priceVariationCollection->build($product, $nostoProduct, $store)
                 );
             }
+            if ($this->nostoDataHelper->isTagDatePublishedEnabled($store)) {
+                $nostoProduct->setDatePublished($product->getCreatedAt());
```

Comment:
We have to catch the exception here

Reply:
That's already wrapped in a try-catch block. I'll catch the `NostoException` one to avoid nesting.